paths_cli/tests/wizard/test_volumes.py

```diff
@@ -0,0 +1,138 @@
+import pytest
+from unittest import mock
+
+from paths_cli.tests.wizard.mock_wizard import mock_wizard
+
+from paths_cli.wizard.volumes import (
+    _vol_intro, intersection_volume, union_volume, negated_volume,
+    cv_defined_volume, volumes, SUPPORTED_VOLUMES
+)
+
+import openpathsampling as paths
+from openpathsampling.experimental.storage.collective_variables import \
+        CoordinateFunctionCV
+
+from openpathsampling.tests.test_helpers import make_1d_traj
+
+def _wrap(x, period_min, period_max):
+    # used in testing periodic CVs
+    while x >= period_max:
+        x -= period_max - period_min
+    while x < period_min:
+        x += period_max - period-min
+    return x
+
+
+@pytest.fixture
+def volume_setup():
+    cv = CoordinateFunctionCV(lambda snap: snap.xyz[0][0]).named('x')
+    vol1 = paths.CVDefinedVolume(cv, 0.0, 1.0)
+    vol2 = paths.CVDefinedVolume(cv, 0.5, 1.5)
+    return vol1, vol2
+
+@pytest.mark.parametrize('as_state,has_state', [
+    (True, False), (True, True), (False, False)
+])
+def test_vol_intro(as_state, has_state):
+    wizard = mock_wizard([])
+    wizard.requirements['state'] = ('states', 2, float('inf'))
+    if has_state:
+        wizard.states['foo'] = 'placeholder'
+
+    intro = _vol_intro(wizard, as_state)
+
+    if as_state and has_state:
+        assert "another stable state" in intro
+    elif as_state and not has_state:
+        assert "You'll need to define" in intro
+    elif not as_state:
+        assert intro is None
+    else:
+        raise RuntimeError("WTF?")
```

Comment:
I prefer to explicitly test the condition `not as_state`, just in case the code around it changes. I tossed in the final `else: WTF?` because I'm always annoyed if a series of `elif`s don't have a catch-all `else` at the end. Since the condition should be impossible, I'm not worried about potentially offending anyone (I'll try to keep it mostly PG for users, but I'm not worried about offending readers of the code -- just wait until you see one of the "commands" I'm working on for the next generation of the Wizard!)